runtime/runtime/src/lib.rs

```diff
@@ -2324,19 +2440,111 @@ struct ApplyProcessingReceiptState<'a> {
     local_receipts: VecDeque<Receipt>,
     incoming_receipts: &'a [Receipt],
     delayed_receipts: DelayedReceiptQueueWrapper,
+    pipeline_manager: pipelining::ReceiptPreparationPipeline,
+}
+
+trait MaybeRefReceipt {
+    fn as_ref(&self) -> &Receipt;
 }
 
-impl<'a> ApplyProcessingReceiptState<'a> {
-    /// Obtain the next receipt that should be executed.
-    fn next_local_receipt(&mut self) -> Option<Receipt> {
-        self.local_receipts.pop_front()
+impl MaybeRefReceipt for Receipt {
+    fn as_ref(&self) -> &Receipt {
+        self
     }
 }
 
+impl<'a> MaybeRefReceipt for &'a Receipt {
+    fn as_ref(&self) -> &Receipt {
+        *self
+    }
+}
+
+/// Schedule a one receipt for contract preparation.
+///
+/// The caller should call this method again after the returned number of receipts from `iterator`
+/// are processed.
+fn schedule_contract_preparation<'b, R: MaybeRefReceipt>(
+    pipeline_manager: &mut pipelining::ReceiptPreparationPipeline,
+    state_update: &TrieUpdate,
+    mut iterator: impl Iterator<Item = R>,
+) -> Option<usize> {
```

Comment:
Adding 

```
if true {
    return None;
}
```

here in principle disables the whole pipelining ting and no differing hashes are seen.